### ISSUE TITLE
Improve experience for visually impaired users

### ISSUE THREAD

**Issue by Jermolene** [Jermolene/TiddlyWiki5#1682]
As raised in [this thread](https://groups.google.com/d/topic/tiddlywiki/gDDaJfQDuwk/discussion), it would be desirable to improve the usability of TiddlyWiki for visually impaired users by introducing the following features:

* New high contrast palettes, in both dark-on-light and light-on-dark variants
* New themes that simplify the main page so that it displays well at extremely large font sizes

In order to make these changes effective we'll need to work closely with visually impaired users who can try it out and give us feedback. There are many issues that it would be helpful to understand:

* Is there a preference for icons or text labels, or a combination?
* Is there a preference for using a keyboard over a mouse or trackpad?

**Comment by roma0104** [comment]
Here is some quick reading for anyone interested in this:
Much is about actual text design but there are some some about machine/human interface:

* http://universaldesign.ie/Technology-ICT/Irish-National-IT-Accessibility-Guidelines/Application-software/Guidelines-for-Applications-Software-Priority-1/1-5-Do-not-require-use-of-a-pointing-device/
* http://guidelines.usability.gov/
* http://www.si.edu/Content/Accessibility/Publication-Guidelines.pdf

What I have learn from a couple hours of reading, on your last two points @Jermolene:
* For languages that use the Roman alphabet, words are the preferred over images.
* Depending on the user's needs for the input type.  The application should be totally accessible with a keyboard; TW5 seems to be when I was playing, though it took much time to tab through all the interaction points.  Perhaps, when an input is not in focus, certain keys perform actions.  Though, we would want to avoid modifier keys.

**Comment by sukima** [comment]
> Pressing <kbd>space</kbd> will make TW5 listen, followed by a <kbd>x</kbd> will close the topmost tiddler.

Oh this just made my little [Vim](http://www.vim.org/) :heart: flutter with joy. The *make TW5 listen* part practically described a keyboard based modal interface just like with Vi / Vim's philosophy. One could reword it to fit the same:

> Press <kbd>Esc</kbd> to return to normal mode, followed by an <kbd>x</kbd> to delete the current character.

The connection is uncanny!

**Comment by Jermolene** [comment]
Hi @sukima I'm very keen to have a really decent keyboard interface (I reposted some early sketches to the group - https://github.com/Jermolene/TiddlyWiki5/commit/c2e99647d02f0bc8d16feac6039b49afb445bc14#diff-12b7f28a28eef72270ce72ba360bfa9c).

Also, thanks for reminding me of the `<kbd>` element:

https://github.com/Jermolene/TiddlyWiki5/commit/6942efa6289c9f1d99ea9f63972aa314ac06a8b1

**Comment by tobibeer** [comment]
I think there's plenty room, especially for improving the usability of the editing experience ...via keyboard rather than mouse. The add-this-popups have their ways of impinging on usability. In TWc they were not only accessed by pressing the DOWN key, but also got focus that way so that one could select an option, let alone ESC out of the popup.

**Comment by cary-rowen** [comment]
Hi @Jermolene 

I'm a visually impaired user of TW and I'm amazed and excited about TW's Accessibility, Thank you, this is great!
Currently I want to know, as a visually impaired user using NVDA, how to add plugins, is there any other option for visually impaired users besides drag and drop?

Grateful

**Comment by abesamma** [comment]
Hi @Rowen, it's good to hear that TiddlyWiki is accessible to you! It's a
credit to the developers. There is still more we can do as a community I'm
sure and I'm interesting in hearing more in this regard.

With respect to your question: Like ordinary tiddlers, plugins can be
imported via the import button as json files or tid files. If the plugin
contains JavaScript files, then after importing, TiddlyWiki will ask you to
save the wiki and reload it in the usual manner. Otherwise, the plugin may
work immediately without prompting you for a refresh.

Hope that helps. Let us know if you have any other questions or concerns.

**Comment by AllenChen-Xingan** [comment]
S a screen reader user as well, it’s a great news to hear that we can import plug-ins via no drag and drop. however for most plug-ins from Github or other tid projects, how to translate plug-in links to JSON /tid files

**Comment by abesamma** [comment]
@AllenChen-Xingan if the project has a demo tiddlywiki (and that is usually the case) that shows the plugin in action, exporting the plugin tiddler should do the trick, since all the necessary shadow tiddlers composing the plugin is packed into this one plugin tiddler. So you can simply export them as json files, then import them into your tiddler.

I recommend visiting [TiddlyLinks](https://links.tiddlywiki.com/) for links to these tiddlywiki demo pages.

**Comment by AllenChen-Xingan** [comment]
@abesamma great thanks for your response. i was trying to import this [relink](https://flibbles.github.io/tw5-relink/) plugin. i clicked on the "drag&drop" link then clicked more and select export as Json file. next importing to my own tiddly file but i still did not see the plug-in in the plugins list.

**Comment by pmario** [comment]
@AllenChen-Xingan ... Just to be sure: The relink plugin requires a "save" and "browser reload" after you did install it.

**Comment by cary-rowen** [comment]
thanks @pmario @abesamma 

As Allen said, I tried importing 'tiddlers.json' and after clicking the "save" button my Chrome browser downloaded another html file, unfortunately I didn't find the plugin I just installed in this new file.
Of course, no new plugins were found in the previous HTML.
btw, as a screen reader user, The way I would prefer might be to copy the plugin's URL and paste it in my TW to install the plugin.

All the best

Cary

**Comment by pmario** [comment]
>btw, as a screen reader user, The way I would prefer might be to copy the plugin's URL and paste it in my TW to install the plugin.

@cary-rowen ... Thanks for the info. 

But I think the URLs for plugins are hard to understand, if read by a screen reader. For example https://tiddlywiki.com/#%24%3A%2Fplugins%2Ftiddlywiki%2Finternals

Or do you have an other URL in mind?

**Comment by cary-rowen** [comment]
Hi @pmario 
I should be very clear about the source of the URL, so there is no need to judge what plugin this is by the URL.
I'm not sure I fully understand your doubt.

Best
